### PR TITLE
Visual Test Addon Onboarding behind query param

### PR DESCRIPTION
**Pull request by weeksling**
Resolve AP-4116, AP-4126, AP-4127, AP-4133, AP-4115, AP-4110

This adds addon onboarding, with steps:
 - Initital Build flow - to prompt brand new projects to create a first build baseline before testing changes.
 - Make a Change flow - force the user to create a local change, and run a local build to test for changes. (will rerun if the build does not have any changes caught).
 - Done screen - displays if the last build did have changes, and prompts user to continue to guided walkthrough.
 - Guided Tour - once a build exists with changes, guide the user through the story filters, review screens, accepting a story, and rerunning builds.

To enable the onboarding, add the `&vtaOnboarding=true` query param to your storybook url.

To test, you will want to try this out in these different scenarios:
 1. Create a brand new project on Chromatic (no builds) and add the addon. Onboarding will guide you through creating very first build, making a change, and then the guided tour.
 2. Add the addon to a project with existing builds on Chromatic. Onboarding will skip ahead to make a change and running second build.
 3. Opening a branch with an existing build that has changes. Onboarding will show you the Done screen, and then begin the guided tour (this will be updated to skip ahead to the guided tour without the done screen. Was kept to get a version of the onboarding out - this is a less common edge case).
 

<details>
  <summary>📦 Published PR as canary version: <code>1.0.1--canary.144.53a44d0.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @chromaui/addon-visual-tests@1.0.1--canary.144.53a44d0.0
  # or 
  yarn add @chromaui/addon-visual-tests@1.0.1--canary.144.53a44d0.0
  ```
</details>

### CODE REVIEW

File: public/Snapshot-Preview.png
Comment (ghengeveld) [inline public/Snapshot-Preview.png:1]: This should be the actual component rather than our barchart.

**Comment by paulelliott** [comment]
Invoking the circuit breaker on this PR as it's gone on well past the end date of the project.

File: src/Panel.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {
 import { Project } from "./gql/graphql";
 import { Authentication } from "./screens/Authentication/Authentication";
 import { GitNotFound } from "./screens/GitNotFound/GitNotFound";
+import { GuidedTour } from "./screens/GuidedTour/GuidedTour";
```
Comment (JonathanKolnik) [inline src/Panel.tsx:19]: This seems unused

File: src/screens/GuidedTour/GuidedTour.tsx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+          ...storybookLayout,
+          resizerPanel: { x: window.innerWidth - 500 },
+        })
+      );
+      document.location.reload();
+    }
+  }, [managerApi]);
+
+  const [showConfetti, setShowConfetti] = React.useState(false);
+  // This will just be shown by default for now. Need to figure out when we should display it.
```
Comment (JonathanKolnik) [inline src/screens/GuidedTour/GuidedTour.tsx:89]: Is this still a todo for this PR?

File: src/screens/Onboarding/Onboarding.stories.tsx [hunk 3]
```diff
@@ -43,46 +60,237 @@ export const Default: Story = {
...
+      "https://www.figma.com/file/GFEbCgCVDtbZhngULbw2gP/Visual-testing-in-Storybook?type=design&node-id=352-258984&t=3EAIRe8423CpOQWY-4"
+    ),
+  },
 };
 
-export const Done: Story = {
-  parameters: withFigmaDesign(
-    "https://www.figma.com/file/GFEbCgCVDtbZhngULbw2gP/Visual-testing-in-Storybook?type=design&node-id=304-318693&t=3EAIRe8423CpOQWY-4"
-  ),
+// TODO: This design for an error in the Onboarding is incomplete
```
Comment (JonathanKolnik) [inline src/screens/Onboarding/Onboarding.stories.tsx:275]: Just highlighting this in case it is meant to be part of this PR

File: src/screens/Onboarding/Onboarding.tsx [hunk 4]
```diff
@@ -20,33 +35,307 @@ const ProjectQuery = gql`
...
+          <Button onClick={onSkip} link>
+            Skip walkthrough
+          </Button>
+        </Stack>
+      </Container>
+    );
+  }
+
+  if (
+    showInitialBuild &&
```
Comment (JonathanKolnik) [inline src/screens/Onboarding/Onboarding.tsx:135]: This is more of a nit and doesn't have to be addressed in this PR but the logic for what to show could be refactored a bit to make it more readable

**Review by JonathanKolnik — COMMENTED** [review]
The comments make it a bit unclear if this is a PR that is ready to merge or not

**Comment by JonathanKolnik** [comment]
I'm new to reviewing this repo, if there are easy ways to get it into the right state to QA please let me know!

File: src/screens/Onboarding/Onboarding.stories.tsx [hunk 5]
```diff
@@ -43,46 +60,237 @@ export const Default: Story = {
...
+      ...INITIAL_BUILD_PAYLOAD,
+    },
+    showInitialBuildScreen: true,
+  },
   parameters: withFigmaDesign(
     "https://www.figma.com/file/GFEbCgCVDtbZhngULbw2gP/Visual-testing-in-Storybook?type=design&node-id=304-318374&t=3EAIRe8423CpOQWY-4"
   ),
 };
 
+// This story is broken until a soltion is added to show
```
Comment (weeksling) [inline src/screens/Onboarding/Onboarding.stories.tsx:80]:
```suggestion
```

File: src/screens/Onboarding/Onboarding.tsx (hunk 4 above)
Comment (weeksling) [reply src/screens/Onboarding/Onboarding.tsx:135]: I will try to improve this in a follow-up. I'm not ready to rewrite the navigation logic yet.

File: src/screens/Onboarding/Onboarding.stories.tsx (hunk 3 above)
Comment (weeksling) [reply src/screens/Onboarding/Onboarding.stories.tsx:274]: I will remove this story, as this story does technically run. There was no design for it, but if there was an error, it will display the screen given. If an improvement is necessary we will find that in design review.

File: src/screens/Onboarding/Onboarding.tsx (hunk 4 above)
Comment (weeksling) [reply src/screens/Onboarding/Onboarding.tsx]: I plan to handle more improvements in a follow-up pr, but in the latest changes I have at least simplified the if conditions to make it a little easier to refactor eventually, and removed the no screen case, as there should now be no reasons for it to mistakenly display no screen.

**Comment by JonathanKolnik** [comment]
Notes from this case
>2. Add the addon to a project with existing builds on Chromatic. Onboarding will skip ahead to make a change and running second build.

- [ ] missing images and state change loss if you "resize" in this case I opened devtools and that triggered it 

https://github.com/chromaui/addon-visual-tests/assets/1164060/c7853fe5-ce3b-498a-8544-03eccce686cf


- [ ] I still saw the flicker while building the first build (maybe that's a separate ticket already)
- [ ] Should this view have a disabled bg? 
image

**Review by JonathanKolnik — APPROVED** [review]
All three QA scenarios check out 👍

**Comment by weeksling** [comment]
Fixed the image issue and pushed new release 0.0.127-canary.0 - will look at others early Monday. This should otherwise be ready for testing. Will share version in the slack channel.
